components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -164,6 +141,30 @@ export default class ChannelDetails extends React.Component<ChannelDetailsProps,
 
         await Promise.all(actionsToAwait);
     }
+
+    private updateChannelPermissions() {
```

Comment:
NIT: Maybe we can rename this to `updateUseChannelMentions` or something since currently it's only updating those and not the other permisisons. Thoughts? More specific but don't mind either way.

Reply:
I have renamed it to `restrictChannelMentions` since that is what its doing at its core - restricting channel mentions based on the create post value